homework.js

```diff
@@ -0,0 +1,162 @@
+    return wordsArray.join('-');
+}
+
+/*
+  Function that will count up to a number and back down and return a string of the climb 
+  countUpAndDown(3)  should return "1 2 3 2 1"
+*/
+const countUpAndDown = function (num) {
+    let result = []
+    for (let i = 1; i <= num; i++) {
+        result.push(i);
+    }
+    for (let i = num; i >= 1; i--) {
+        result.push(i);
+    }
+    return result.join(" ");
+}
+
+/*
+  Write a function that will tell you all of the words in an array that contain the letter `a` 
+  wordsWithA(['cat', 'rabbit', 'dog', 'frog'])  should return ['cat', 'rabbit']
+*/
+const wordsWithA = function (wordsArray) {
+    const resultArrayWithA = []
+    for (let i = 0; i < wordsArray.length; i++) {
+        if (wordsArray[i].indexOf('a') >= 0) {
+            resultArrayWithA.push(wordsArray[i])
+        }
+    }
+    return resultArrayWithA;
+}
+
+/*
+  Write a function that will tell you all of the words in an array that contain a specified letter 
+  wordsWithLetter("g", ['cat', 'rabbit', 'dog', 'frog'])  should return ['dog', 'frog']
+*/
+const wordsWithLetter = function (letter, wordsArray) {
+    const resultArrayWithUsrInput = []
+    for (let i = 0; i < wordsArray.length; i++) {
+        if (wordsArray[i].indexOf(letter) >= 0) {
+            resultArrayWithUsrInput.push(wordsArray[i])
+        }
+    }
+    return resultArrayWithUsrInput;
+}
+
+/*
+  Function that returns the longest word in sentence 
+  longestWord("The cat in the house")  should return "house"
+*/
+const longestWord = function (sentence) {
+    const words = sentence.split(" ");
+    let longestWord = "";
+    for (let i = 0; i < words.length; i++) {
+        if (words[i].length > longestWord.length) {
+            longestWord = words[i];
+        }
+    }
+    return longestWord;
+}
+
+/*
+  Function that returns the largest even number 
+  largestEvenNumber([1,2,3,10,4,7,0])  should return "10"
+*/
+const largestEvenNumber = function (numbersArray) {
+    let largestEven = 0;
+    for (let i = 0; i < numbersArray.length; i++) {
+        if ((numbersArray[i] % 2 === 0) && (numbersArray[i] > largestEven)) {
+            largestEven = numbersArray[i];
+        }
+    }
+    return largestEven;
+}
+
+/*
+  Extra Practice
+*/
+const wordLetters = ['G', 'O', 'A', 'T'];
+const guessedLetters = ['_', '_', '_', '_'];
+
+const guessLetter = function (letter = '') {
+    letter = letter.toUpperCase();
+
+    for (let i = 0; i < wordLetters.length; i++) {
+        if (wordLetters[i] == letter) {
+            guessedLetters[i] = letter;
+            correctGuess = true;
+        }
+        if (guessedLetters[i] == '_') {
+            tryAgain = true;
+        }
+    }
+    if (correctGuess) {
+        console.log('Correct, ' + guessedLetters.join(''));
+        if (!tryAgain) {
+            console.log('You Win, G O A T');
+        }
+    } else {
+        console.log('Incorrect, ' + guessedLetters.join(''));
+    }
+}
```

Comment:
Good work, but the player never loses. and incorrect guesses are counted as correct